app/models/miq_queue.rb

```diff
@@ -278,12 +278,35 @@ def self.put_or_update(find_options)
   #   puts a new item on the queue.  If the item was found, it will not be
   #   changed, and will be yielded to an optional block, generally for logging
   #   purposes.
+
+  # Return task.id
   def self.put_unless_exists(find_options)
-    put_or_update(find_options) do |msg, item_hash|
-      ret = yield(msg, item_hash) if block_given?
-      # create the record if the original message did not exist, don't change otherwise
-      ret if msg.nil?
+    find_options = default_get_options(find_options)
```

Comment:
we'll want to make sure that this doesn't blow away the `create_with` and that this doesn't add anything that we don't want in the `put`.

(I may be wrong that we can get rid of `conds`